### PR TITLE
GDPR-145 Providing ability to submit ad hoc deletion requests

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/dto/AdHocReferralRequest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonInclude(NON_NULL)
+public class AdHocReferralRequest {
+
+    @JsonProperty("offenderIdDisplay")
+    private String offenderIdDisplay;
+
+    @JsonProperty("batchId")
+    private Long retentionCheckId;
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/dto/AdHocReferralRequest.java:21]: Should this be batchId?  meh

File: src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/sqs/DataComplianceAwsEventPusher.java [hunk 2]
```diff
@@ -88,6 +90,16 @@ public void requestFreeTextMoratoriumCheck(final OffenderNumber offenderNo,
                 new FreeTextSearchRequest(offenderNo.getOffenderNumber(), retentionCheckId, regex)));
     }
 
+    @Override
+    public void requestAdHocReferral(final OffenderNumber offenderNo, final Long batchId) {
+
+        log.debug("Requesting deletion referral for offender: '{}' and batch: '{}'",
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/sqs/DataComplianceAwsEventPusher.java:96]: perhaps reference the fact it is an ad-hoc in the log message....meh

File: src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/sqs/DataComplianceNoOpEventPusher.java [hunk 3]
```diff
@@ -42,4 +42,10 @@ public void requestFreeTextMoratoriumCheck(final OffenderNumber offenderNo,
         log.warn("Pretending to push free text moratorium check '{}/{}' to queue",
                 offenderNo.getOffenderNumber(), retentionCheckId);
     }
+
+    @Override
+    public void requestAdHocReferral(OffenderNumber offenderNo, Long batchId) {
+        log.warn("Pretending to request referral for offender: '{}' and batch: '{}'",
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/events/publishers/sqs/DataComplianceNoOpEventPusher.java:48]: Same for log message here? meh

File: src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralService.java [hunk 4]
```diff
@@ -30,6 +34,19 @@
     private final OffenderDeletionBatchRepository batchRepository;
     private final RetentionService retentionService;
     private final ReferralResolutionService referralResolutionService;
+    private final DataComplianceEventPusher deletionGrantedEventPusher;
+
+    public void handleAdHocDeletion(final AdHocOffenderDeletion event) {
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralService.java:39]: Not sure you really need the type of event in the method name as this is determined by the type being passed in...meh

**Comment by brightonsbox** [comment]
Thanks Matt, I've added the changes locally so I'll push in the next PR

**Pull request merged**